### PR TITLE
Add analytics tracking to Alert Box link clicks

### PR DESCRIPTION
**Pull request by bsmartin-ep**
Standardize data layer events and variables to track alert box components across www.va.gov

re:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/14489

### CODE REVIEW

**Comment by bkjohnson** [comment]
I'll hold off on reviewing this until #35 is merged, since both of these PRs share the new analytics helper function.

File: src/components/AlertBox/AlertBox.unit.spec.jsx
Comment (bkjohnson) [inline src/components/AlertBox/AlertBox.unit.spec.jsx:124]: Since the component is set up with the event handler on the wrapping div, can we have a test that also checks that the event is triggered when a link in the `headline` is clicked?

File: src/components/AlertBox/AlertBox.unit.spec.jsx
Comment (bkjohnson) [inline src/components/AlertBox/AlertBox.unit.spec.jsx:163]: Since the `expect` checking is done in the event handler, it isn't really obvious what we expect if we are just reading this portion of the test. Because of that, can you try to rewrite it so that it resembles:

```js
      const testLink = wrapper.find('a');
      testLink.simulate('click');
      
      expect(eventHandler.calledWith(eventArgs)).to.be.true)
```

This is assuming that `eventHandler` was set up previously with something like `sinon.spy()`.

File: src/components/AlertBox/AlertBox.unit.spec.jsx
Comment (bsmartin-ep) [reply src/components/AlertBox/AlertBox.unit.spec.jsx]: Done!

File: src/components/AlertBox/AlertBox.unit.spec.jsx
Comment (bsmartin-ep) [reply src/components/AlertBox/AlertBox.unit.spec.jsx]: Thanks for taking care of the eslint stuff this AM.

Let me know if this is what you were thinking as far as testing:

https://github.com/department-of-veterans-affairs/component-library/blob/8258a488b1bc3212bc0bed009396aea0c0382541/src/components/AlertBox/AlertBox.unit.spec.jsx

File: package.json
Comment (bsmartin-ep) [inline package.json:3]: This feel like a patch / fix to you?  Or bump to 2.1.0?

File: src/components/AlertBox/AlertBox.unit.spec.jsx
Comment (bsmartin-ep) [reply src/components/AlertBox/AlertBox.unit.spec.jsx]: Sure thing.  Here's what I just committed...

```js
const handleAnalyticsEvent = sinon.spy();

global.document.body.addEventListener(
  'component-library-analytics',
  handleAnalyticsEvent,
);

// Click link in headline
const testLink = wrapper.find('.usa-alert-heading a');
testLink.simulate('click');

global.document.body.removeEventListener(
  'component-library-analytics',
  handleAnalyticsEvent,
);

expect(
  handleAnalyticsEvent.calledWith(
    sinon.match.has('detail', {
      componentName: 'AlertBox',
      action: 'linkClick',
      details: {
        clickLabel: 'with a link',
        headline: HeadlineWithLink,
        status: 'info',
        backgroundOnly: true,
        closeable: false,
      },
    }),
  ),
).to.be.true;
```

File: src/components/AlertBox/AlertBox.unit.spec.jsx
Comment (bkjohnson) [inline src/components/AlertBox/AlertBox.unit.spec.jsx:170]: This is cool!

**Review by bkjohnson — APPROVED** [review]
:+1:

**Pull request merged**
